lib/spack/spack/compilers/aocc.py

```diff
@@ -118,3 +119,37 @@ def f77_version(cls, f77):
     @property
     def stdcxx_libs(self):
         return ('-lstdc++', )
+
+    @property
+    def cflags(self):
+        return self.__handle_default_flag_addtions()
+
+    @property
+    def cxxflags(self):
+        return self.__handle_default_flag_addtions()
+
+    @property
+    def fflags(self):
+        return self.__handle_default_flag_addtions()
+
+    def __handle_default_flag_addtions(self):
```

Comment:
Let's not introduce double underscore functions in Spack, we are trying to avoid them due to the weird semantics :slightly_smiling_face: 
```suggestion
    def _handle_default_flag_addtions(self):
```